### DISCUSSION TITLE
WITH ORDINALITY

### DISCUSSION THREAD

**Discussion by ArianYN** [sqlpage/SQLPage#491] [Q&A] [answered]
Is WITH ORDINALTY not a feature in SQLPage?

I cannot seem to get it to work, no matter what I do...

**Comment by lovasoa** [comment]
Hello and welcome to SQLPage !
  Can you please post your sql query, the error message you are getting, and the database you are using ?

**Reply by ArianYN** [reply]
I am trying to run this 
`JOIN LATERAL (
    SELECT e.value
    FROM jsonb_array_elements(j1.pnet_nodes) WITH ORDINALITY AS e(value, ord)
    WHERE e.ord > 1 
) AS filtered_e ON true`

But the error i am facing is 

`SQLPage found a syntax error on line 296, character 44:
) WITH ORDINALITY AS e ( value , ord ) WHERE e . ord > 1 ) AS filtered_e ON true WHERE ( j1 . pnet_nodes -> 0 ->> 'port' ) :: `

sql parser error: Expected ), found: WITH at Line: 296, Column 46
SQLPage could not parse and prepare this SQL statement

**Reply by lovasoa** [reply, marked as answer]
Thanks for the report ! I opened an issue upstream in order to track and fix the bug in our sql parser: https://github.com/sqlparser-rs/sqlparser-rs/issues/1336

In the meantime, you can use `row_number()` to emulate the functionality:

```sql
join lateral (
  select value, row_number() over () as ord
  from jsonb_array_elements(j1.pnet_nodes)
) as filtered_e on true
where ord>1
```
[demo](https://sqliteonline.com/#sqltext=%23url-sqlite%3Ddb-postgres%0D%0A%23tab-name%3DPostgreSQL%0D%0Adrop%20table%20if%20exists%20t%3B%0Acreate%20table%20t%20as%20(%0A%20%20select%201%20as%20id%2C%20'%5B87%2C88%5D'%3A%3Ajsonb%20as%20js%20UNION%20ALL%0A%20%20select%202%2C%20'%5B55%2C56%2C57%5D'%0A)%3B%0A%0A%0Aselect%20*%20%0Afrom%20t%0Ajoin%20lateral%20(%0A%20%20select%20value%2C%20row_number()%20over%20()%20as%20ord%0A%20%20from%20jsonb_array_elements(T.js)%0A)%20as%20filtered_e%20on%20true%0Awhere%20ord%3E1%3B)

**Reply by ArianYN** [reply]
Okay, thanks 👍
